### PR TITLE
feat: [STO-5759]: Report STO Usage Job

### PR DESCRIPTION
**Pull request by ahokinson**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48261)

### CODE REVIEW

**Review by r0610205 — DISMISSED** [review]

File: 315-sto-manager/app/src/main/java/io/harness/app/telemetry/STOTelemetryPublisher.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      for (STOUsage usage : allUsage.usage) {
+        if (EmptyPredicate.isNotEmpty(usage.accountId) && !usage.accountId.equals(GLOBAL_ACCOUNT_ID)) {
+          HashMap<String, Object> map = new HashMap<>();
+          map.put(GROUP_TYPE, ACCOUNT);
+          map.put(GROUP_ID, usage.accountId);
+          map.put(ACCOUNT_DEPLOY_TYPE, System.getenv().get(DEPLOY_VERSION));
+          map.put(COUNT_ACTIVE_DEVELOPERS, usage.developerCount);
+          map.put(COUNT_SCANS, usage.scanCount);
+          telemetryReporter.sendGroupEvent(usage.accountId, null, map, Collections.singletonMap(ALL, true),
+              TelemetryOption.builder().sendForCommunity(true).build());
```
Comment (r0610205) [inline 315-sto-manager/app/src/main/java/io/harness/app/telemetry/STOTelemetryPublisher.java:57]: We need to figure out what's the correct value here

File: 315-sto-manager/app/src/main/java/io/harness/app/telemetry/STOTelemetryRecordsJob.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  @Inject @Named("stoTelemetryPublisherExecutor") protected ScheduledExecutorService executorService;
+  @Inject STOTelemetryPublisher publisher;
+
+  public void scheduleTasks() {
+    long initialDelay = 1800;
+
+    try {
+      log.info("STOTelemetryRecordsJob scheduler starting");
+      executorService.scheduleAtFixedRate(() -> publisher.recordTelemetry(), initialDelay, DAILY, TimeUnit.SECONDS);
+      log.info("CiTelemetryRecordsJob scheduler started");
```
Comment (r0610205) [inline 315-sto-manager/app/src/main/java/io/harness/app/telemetry/STOTelemetryRecordsJob.java:31]: STO

**Comment by ahokinson** [comment]
trigger uts

**Comment by ahokinson** [comment]
trigger uts

**Review by r0610205 — DISMISSED** [review]

**Review by atmollo-harness — APPROVED** [review]

**Review by r0610205 — APPROVED** [review]

**Pull request merged**